### PR TITLE
add app management capabilities

### PR DESCRIPTION
**Pull request by dexhorthy**
$ go run ./cli app create dex-4
	ID                             NAME     SLUG             SCHEDULER
	1qUVWDuO9XsaIZRuqOnGS42L3uz    dex-4    dex-4-manatee    kots

	$ go run ./cli app delete dex-4-manatee
	ID                             NAME     SLUG             SCHEDULER
	1qUVWDuO9XsaIZRuqOnGS42L3uz    dex-4    dex-4-manatee    kots
	Delete the above listed application? There is no undo:   no
	Error: prompt declined
	exit status 1

	$ go run ./cli app delete dex-4-manatee
	ID                             NAME     SLUG             SCHEDULER
	1qUVWDuO9XsaIZRuqOnGS42L3uz    dex-4    dex-4-manatee    kots
	Delete the above listed application? There is no undo:   yes

	$ go run ./cli app ls dex-
	ID                             NAME                      SLUG                    SCHEDULER
	1gpU6hFrwvXQvdU7CtG5ZTGRuy3    kots-dex-integration      kots-dex-integration    kots
	1mIxSexfFN9SYPdEo6o4Ko1PQ5d    kots-dex-2                kots-dex-2-longhorn     kots
	1XDWkWVtgvr4A20TREpCHEwRAun    kots-dex-the third one    kots-dex-2              kots

	$ go run ./cli app ls dex-4
	ID    NAME    SLUG    SCHEDULER

### CODE REVIEW

File: cli/cmd/app_create.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+func (r *runners) createApp(_ *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return errors.New("missing app name")
+	}
+	appName := args[0]
+
+	kotsRestClient := kotsclient.VendorV3Client{HTTPClient: *r.platformAPI}
+
+
+	app, err := kotsRestClient.CreateKOTSApp(appName)
```
Comment (dexhorthy) [inline cli/cmd/app_create.go:33]: stole this logic from integration tests

File: cli/cmd/app_delete.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		RunE:         r.deleteApp,
+		SilenceUsage: true,
+	}
+	parent.AddCommand(cmd)
+	cmd.Flags().BoolVarP(&r.args.deleteAppForceYes, "force", "f", false, "Skip confirmation prompt. There is no undo for this action.")
+
+	return cmd
+}
+
+func (r *runners) deleteApp(_ *cobra.Command, args []string) error {
```
Comment (dexhorthy) [inline cli/cmd/app_delete.go:26]: delete an app, prompt for confirm

File: cli/cmd/app_delete.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+func promptConfirmDelete() (string, error) {
+
+	prompt := promptui.Prompt{
+		Label:     "Delete the above listed application? There is no undo:",
+		Templates: templates,
+		Default:   "",
+		Validate: func(input string) error {
+			if input == "no" {
+				return nil
```
Comment (dexhorthy) [inline cli/cmd/app_delete.go:86]: just in case they don't know how to ctrl+c ??

File: cli/print/apps.go [hunk 4]
```diff
@@ -4,25 +4,21 @@ import (
...
 {{ range . -}}
-{{ .ID }}	{{ .Name }}	{{ .Scheduler }}
+{{ .ID }}	{{ .Name }}	{{ .Slug}}	{{ .Scheduler }}
 {{ end }}`
 
 var appsTmpl = template.Must(template.New("apps").Funcs(funcs).Parse(appsTmplSrc))
 
-func Apps(w *tabwriter.Writer, apps []apps.AppAndChannels) error {
-	as := make([]map[string]interface{}, len(apps))
+func Apps(w *tabwriter.Writer, apps []types.AppAndChannels) error {
```
Comment (dexhorthy) [inline cli/print/apps.go:17]: this function was unused

File: client/app.go [hunk 5]
```diff
@@ -42,6 +47,7 @@ func (c *Client) ListApps() ([]types.AppAndChannels, error) {
 	}
 
 	apps = append(apps, shipApps...)
+	apps = append(apps, kotsApps...)
```
Comment (dexhorthy) [inline client/app.go:50]: 🙄

File: cli/cmd/app_delete.go (hunk 3 above)
Comment (laverya) [reply cli/cmd/app_delete.go:86]: uh
doesn't this have the same result as typing "yes", by returning null?

There's already the "!= yes" case, and IMO this should be handled there

File: cli/test/kots_app_test.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+	"github.com/replicatedhq/replicated/pkg/kotsclient"
+	"github.com/replicatedhq/replicated/pkg/platformclient"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+var _ = Describe("kots apps", func() {
+	t := GinkgoT()
```
Comment (laverya) [inline cli/test/kots_app_test.go:17]: why ginkgo and not pact?

**Review by laverya — APPROVED** [review]

File: cli/cmd/app_delete.go (hunk 3 above)
Comment (dexhorthy) [reply cli/cmd/app_delete.go:86]: it passes validation, but the string returned in the body of the cmd handler will be "no" so the operation won't proceed. We double check in the body to make sure it was "yes" before actually deleting

File: cli/test/kots_app_test.go (hunk 6 above)
Comment (dexhorthy) [reply cli/test/kots_app_test.go:17]: My interpretation is that pact would be for testing the API client code itself, while ginkgo lets me also test the logic in the cobra command handler...

**Pull request merged**